docs/en/deploy/Running-Alluxio-On-a-HA-Cluster.md

```diff
@@ -0,0 +1,359 @@
+
+### Add/Remove Workers Dynamically
+
+Adding a worker to an Alluxio cluster dynamically is as simple as starting a new Alluxio worker
+process, with the appropriate configuration. In most cases, the new worker's configuration should be the same as all
+the other workers' configuration. Run the following command on the new worker to add
+
+```bash
+./bin/alluxio-start.sh worker SudoMount # starts the local worker
+```
+
+Once the worker is started, it will register itself with the Alluxio master, and become part of the Alluxio cluster.
+
+Removing a worker is as simple as stopping the worker process.
+
+```bash
+./bin/alluxio-stop.sh worker # stops the local worker
+```
+
+Once the worker is stopped, and after
+a timeout on the master (configured by master parameter `alluxio.master.worker.timeout`), the master
+will consider the worker as "lost", and no longer consider it as part of the cluster.
+
+### Add/Remove Masters
+
+In order to add a master, the Alluxio cluster must be in HA mode. If you are running the cluster as
+a single master cluster, you must configure it to be an HA cluster to be able to have more than 1
+master.
+
+#### Alluxio HA cluster with embedded journal
+
+When internal leader election is used, Alluxio masters are determined. Adding or removing
+master nodes requires:
+
+* Shut down the whole cluster
+* Add/remove one or more Alluxio master
+* Format the whole cluster
+* Update the cluster-wide embedded journal configuration
+* Start the whole cluster
+
+Note that all previously stored data and metadata in Alluxio filesystem will be erased.
+
+#### Alluxio HA cluster with Zookeeper and Shared Journal
+
+To add a master to an HA Alluxio cluster, you can simply start a new Alluxio master process, with
+the appropriate configuration. The configuration for the new master should be the same as other masters,
+except that the parameter `alluxio.master.hostname=<MASTER_HOSTNAME>` should reflect the new hostname.
+Once the new master is started, it will start interacting with ZooKeeper to participate in leader election.
+
+Removing a master is as simple as stopping the master process. If the cluster is a single master cluster,
+stopping the master will essentially shutdown the cluster, since the single master is down. If the
+Alluxio cluster is an HA cluster, stopping the leading master will force ZooKeeper to elect a new leading master
+and failover to that new leader. If a standby master is stopped, then the operation of the cluster is
+unaffected. Keep in mind, Alluxio masters high availability depends on the availability on standby
+masters. If there are not enough standby masters, the availability of the leading master will be affected.
+It is recommended to have at least 3 masters for an HA Alluxio cluster.
+
+### Modify Configuration
+
+In order to update the service configuration, you must first [stop the service](#stop-alluxio),
```

Comment:
Maybe include that in HA mode this can be done in a rolling fashion to avoid downtime.

Reply:
Can we? I am not quite sure about the rolling update on master-side configuration.

Reply:
AFAIK this works well in 1.8

Reply:
@apc999 Can you add this part?

Reply:
```suggestion
One benefit of running Alluxio in HA mode is the ability to minimize downtime when updating configurations by using rolling restarts. 
1. Update the master configuration on the master nodes.
1. Restart the leader master (can be determined by running `bin/alluxio leader`)
1. Wait for the leader master to come up successfully as a standby
1. Restart remaining standbys.
1. Verify the configuration update
```